clang/lib/CodeGen/CGExprComplex.cpp

```diff
@@ -310,6 +310,13 @@ class ComplexExprEmitter
         CGF.getContext().getFloatTypeSemantics(ElementType);
     const llvm::fltSemantics &HigherElementTypeSemantics =
         CGF.getContext().getFloatTypeSemantics(HigherElementType);
+    // Check that LongDouble Size > Double Size.
```

Comment:
I'd word this as "Check that the promoted type can handle the intermediate values without overflowing."